### DISCUSSION TITLE
building problem.

### DISCUSSION THREAD

**Discussion by joapatter** [cdcseacave/openMVS#1118] [Q&A] [answered]
Hi, i have followed the instrucctions for building openMVS in my sistem, but i have a problem in this step:
 cmake --build . -j4

this is the problem that im having:
[  1%] Building CXX object libs/Common/CMakeFiles/Common.dir/cmake_pch.hxx.gch
In file included from /usr/local/include/eigen3/Eigen/Core:163,
                 from /home/jpaton/openMVS/libs/Common/Types.h:127,
                 from /home/jpaton/openMVS/libs/Common/Common.h:176,
                 from /home/jpaton/openMVS/make/libs/Common/CMakeFiles/Common.dir/cmake_pch.hxx:5,
                 from <command-line>:
/usr/local/include/eigen3/Eigen/src/Core/util/Meta.h: In function ‘constexpr auto Eigen::internal::index_list_size(T&&)’:
/usr/local/include/eigen3/Eigen/src/Core/util/Meta.h:482:14: error: ‘std::ssize’ has not been declared
  482 |   using std::ssize;
      |              ^~~~~
make[2]: *** [libs/Common/CMakeFiles/Common.dir/build.make:64: libs/Common/CMakeFiles/Common.dir/cmake_pch.hxx.gch] Error 1
make[1]: *** [CMakeFiles/Makefile2:475: libs/Common/CMakeFiles/Common.dir/all] Error 2
make: *** [Makefile:141: all] Error 2

it seems that something is wrong with the compiler, but i cant solve it...

**Comment by cdcseacave** [comment]
all I know it works fine with the setup described in build readme and wiki, and the CI file
pls try following those steps

**Comment by hhl134** [comment]
@joapatter bro, did you solve this problem? I happened to it too

**Comment by ChunqiDai** [comment, marked as answer]
It is because ssize() is supported by gcc10. You should upgrade the version of gcc and g++.

**Reply by cdcseacave** [reply]
thank you
